fw/compile-firmware.md

```diff
@@ -0,0 +1,192 @@
+    ]
+}
+```
+
+We also recommend deleting the other keymaps if you don't use them, as it'll make the action run faster.
+
+Then,
+
+1. Push your changes above to your forked GitHub repository
+1. Look at the GitHub Actions for a new actions run
+1. Wait for the actions run to complete
+1. Inspect the Releases tab on your repository for the latest firmware build
+
+
+# Compiling your firmware using the local command line
+
+You can also compile your firmware through your local command line. This requires you to be familiar with the console, and doing some additional configuration.
+
+## Pre-requisites
+
+### Working QMK environment
+
+Make sure you have a functional QMK environment. See [QMK Docs](https://docs.qmk.fm/#/newbs) for details. At this point, **you don't need** to run `qmk setup`.
+
+### BastardKb QMK fork
+
+Clone the BKB QMK repository, using either github desktop or the command line, and switch to the `bkb-master` branch:
+
+```shell
+git clone https://github.com/bastardkb/bastardkb-qmk
+cd bastardkb-qmk
+git checkout -b bkb-master origin/bkb-master
+```
+
+Now that you've cloned the repository, `cd` into it and set it as the default local QMK repository. You also need to do this if you had a previous, separate QMK installation.
+
+```shell
+qmk config user.qmk_home="$(realpath .)" 
+```
+
+{: .info }
+If you have multiple QMK installations, you will need to manually set the qmk home path again to use the other ones.
+
+That's all you needed to do with the QMK repository.
+From here on, **we will only focus on the userspace repository.**
+
+### QMK Userspace
+
+Next, `cd` into your userspace fork and enable userspace:
+
+```shell
+qmk config user.overlay_dir="$(realpath .)"
+```
+
+## Compiling your firmware
+
+Once in the QMK userspace repository, compiling a keymap works the same as normal:
+
+```shell
+qmk compile -c -kb bastardkb/{keyboard} -km {keymap}
+```
+
+### `{keyboard}` argument
+
+`{keyboard}` corresponds to the physical keyboard you are building the firmware for. It can be one of the following:
+
+- `charybdis/4x6`: the 4x6+5 [Charybdis](https://github.com/bastardkb/charybdis/)
+- `charybdis/3x5`: the 3x5+3 [Charybdis Nano](https://github.com/bastardkb/charybdis/)
+- `charybdis/3x6`: the 3x6+3 [Charybdis Mini](https://github.com/bastardkb/charybdis/)
+- `scylla`: the 4x6+5 [Scylla](https://github.com/Bastardkb/Scylla)
+- `skeletyl`: the 3x5+3 [Skeletyl](https://github.com/Bastardkb/Skeletyl/)
+- `tbkmini`: the 3x6+3 [TBK Mini](https://github.com/Bastardkb/TBK-Mini/)
+- `dilemma/3x5_3`: the 3x5+3 [Dilemma](https://github.com/bastardkb/dilemma/)
+- `dilemma/4x6_4`: the 4x6+4 [Dilemma Max](https://github.com/bastardkb/dilemma/)
+
+### `{keymap}` argument
+
+All keyboards have a `vendor` keymap, which is the one officially supported and shipped with Bastard Keyboards.
+
+To build *any* keymap with Via support, simply add the following argument to the command line: `-e VIA_ENABLE=yes`, e.g.:
```

Comment:
Should make clear that this is not necessary for `vendor` keymaps.
This also explodes it the keymap you're trying it on has more than 4 layers, as that's the default dynamic amount.

That (imho necessary) info bloats this aside to the point where I think it might better to just remove it. It won't be necessary at all because Dactyls used to need it as there was no `via` keymap. But with a `vendor` keymap under our control, that would have via enabled already.